shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/HomeViewModel.kt

```diff
@@ -18,56 +14,78 @@ private const val DEFAULT_PAGE_SIZE = 5
 data class HomeViewState(
     val isLoading: Boolean,
     val headerUiModel: SurveyHeaderUiModel? = null,
-    var surveys: List<SurveyUiModel> = listOf()
+    var surveys: List<SurveyUiModel> = listOf(),
 ) {
-    constructor() : this (true)
+    constructor() : this(true)
 }
 
 class HomeViewModel(
     private val getUserProfileUseCase: GetUserProfileUseCase,
     private val getSurveysUseCase: GetSurveysUseCase,
+    private val getAppVersionUseCase: GetAppVersionUseCase,
     private val dateTime: DateTime,
     private val dateTimeFormatter: DateTimeFormatter,
 ) : BaseViewModel() {
 
     private val _viewState: MutableStateFlow<HomeViewState> = MutableStateFlow(HomeViewState())
-
     val viewState: StateFlow<HomeViewState> = _viewState
 
+    private val _appVersion: MutableStateFlow<String> = MutableStateFlow("")
+    val appVersion: StateFlow<String> = _appVersion
+
     private var previousSelectedIndex = 0
     private var currentPage = 1
 
+    init {
+        fetchData()
+    }
+
     fun fetchData() {
         getProfile()
             .onStart { setStateLoading() }
-            .combine(fetchSurvey(
-                page = currentPage,
-                pageSize = DEFAULT_PAGE_SIZE,
-                isForceLatestData = false
-            )) { user, surveys ->
+            .combine(fetchSurvey(page = currentPage)) { user, surveys ->
                 Pair(user, surveys)
             }
             .onEach {
                 handleFetchUserProfileSuccess(it.first)
                 handleFetchSurveysSuccess(it.second)
             }
             .launchIn(viewModelScope)
+
+        _appVersion.update { getAppVersionUseCase() }
     }
 
     fun fetchMoreSurveysIfNeeded(selectedIndex: Int) {
         if (selectedIndex != _viewState.value.surveys.count() - 1 - LOAD_MORE_THRESHOLD
-            || previousSelectedIndex >= selectedIndex) {
+            || previousSelectedIndex >= selectedIndex
+        ) {
             previousSelectedIndex = selectedIndex
             return
         }
         previousSelectedIndex = selectedIndex
-        fetchSurvey(currentPage, DEFAULT_PAGE_SIZE, false)
+        fetchSurvey(currentPage)
             .onEach {
                 handleFetchMoreSurveySuccess(it)
             }
             .launchIn(viewModelScope)
     }
 
+    fun refresh() {
+        currentPage = 1
+        fetchSurvey(
+            page = currentPage,
+            isForceLatestData = true
+        )
+            .onStart { setStateLoading() }
+            .catch {
+                // TODO: handle error
+            }
```

Comment:
Don't think we need to catch here again. `fetchSurvey` is already catching error isn't it?
```suggestion
```

Reply:
We're ignoring the error in `fetchSurvey()`. We can ignore it here too for now. I'll remove it.